File: Lecture 11/Python Code/Tomato/managers/RestaurantManager.py
class RestaurantManager:
    _instance = None

    def __init__(self):
        self.restaurants = []

    @staticmethod
    def get_instance():
        if RestaurantManager._instance is None:
            RestaurantManager._instance = RestaurantManager()
        return RestaurantManager._instance

    def add_restaurant(self, restaurant):
        self.restaurants.append(restaurant)

    def search_by_location(self, location):
        location = location.lower()
        return [r for r in self.restaurants if r.get_location().lower() == location]

File: Lecture 11/Java Code/Tomato/managers/RestaurantManager.js
export class RestaurantManager {
    constructor() {
        if (RestaurantManager.instance) {
            return RestaurantManager.instance;
        }
        this.restaurants = [];
        RestaurantManager.instance = this;
    }

    static getInstance() {
        if (!RestaurantManager.instance) {
            RestaurantManager.instance = new RestaurantManager();
        }
        return RestaurantManager.instance;
    }

    addRestaurant(r) {
        this.restaurants.push(r);
    }

    searchByLocation(loc) {
        const result = [];
        loc = loc.toLowerCase();
        for (const r of this.restaurants) {
            const rl = r.getLocation().toLowerCase();
            if (rl === loc) {
                result.push(r);
            }
        }
        return result;
    }
}
